lib/chezmoi/script.go

```diff
@@ -121,10 +121,14 @@ func (s *Script) Apply(fs vfs.FS, mutator Mutator, applyOptions *ApplyOptions) e
 	}
 
 	// Write the temporary script file.
-	f, err := ioutil.TempFile("", filepath.Base(s.targetName))
+	pathBase := filepath.Base(s.targetName)
+	pathBase = strings.Replace(pathBase, "#", "*", 1)
```

Comment:
Could you add a comment explaining why this replacement is necessary?

Reply:
Turns out I massively overcomplicated this, which I realized once I started explaining the replacement.  My only defense is that some of this code was written pretty late at night :wink:

(For the record, the point of doing this was to preserve the file extension, which Windows uses to decide if it knows how to execute a file or not. Because of that, `TempFile`'s randomness couldn't just go on the end of the name, which is the default, so I needed a `*` to indicate where it should go.  But `*` is not a legal character in a Windows filename, thus the indirection.  Of course, the much simpler solution is... just put it on the front of whatever the real filename is.)